### ISSUE TITLE
const_item_mutation warning is triggered when modifying an array by pointer

### ISSUE THREAD

**Issue by MSxDOS** [rust-lang/rust#79971]
```Rust
const ARRAY: *mut [u32; 4] = 0x12345678 as _;

pub unsafe fn test(val: u32) {
    (*ARRAY)[0] = val;
}
```
```
Warning: attempting to modify a `const` item
 --> <source>:4:5
  |
4 |     (*ARRAY)[0] = val;
  |     ^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(const_item_mutation)]` on by default
  = note: each usage of a `const` item creates a new temporary; the original `const` item will not be modified
note: `const` item defined here
 --> <source>:1:1
  |
1 | const ARRAY: *mut [u32; 4] = 0x12345678 as _;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
https://github.com/rust-lang/rust/pull/77324 missed that case.
